### DISCUSSION TITLE
Why there is not a main function to run the project?

### DISCUSSION THREAD

**Discussion by oolongkiller** [Telecominfraproject/oopt-gnpy#476] [Q&A] [answered]
Sorry, I have tried to install gnpy by downloading Python packages and just using Pip install. However, I always cannot find the script, which calls 'gnpy-transmission-example'. I refer to the previous discussion and failed to solve this problem. Maybe should I try the docker?
Can you give me some advice?
Thanks a lot.

**Comment by ojnas** [comment, marked as answer]
It's not clear what you are asking. There is no script which calls `gnpy-transmission-example`. `gnpy-transmission-example` is an autogenerated script, which just calls the `transmission_main_example` function in `cli_examples.py`:

https://github.com/Telecominfraproject/oopt-gnpy/blob/master/gnpy/tools/cli_examples.py#L111

The `gnpy-transmission-example` script itself is normally installed in the bin directory in your Python virtual environment but its content is not very interesting:

```
# PBR Generated from 'console_scripts'

import sys

from gnpy.tools.cli_examples import transmission_main_example


if __name__ == "__main__":
    sys.exit(transmission_main_example())
```

**Reply by oolongkiller** [reply]
Thanks a lot !!!
